components/mdns/mdns.c

```diff
@@ -3510,22 +3520,28 @@ void mdns_parse_packet(mdns_rx_packet_t *packet)
 
 #ifndef CONFIG_MDNS_SKIP_SUPPRESSING_OWN_QUERIES
     // Check if the packet wasn't sent by us
+#ifdef CONFIG_MDNS_IPV4
     if (packet->ip_protocol == MDNS_IP_PROTOCOL_V4) {
         esp_netif_ip_info_t if_ip_info;
         if (esp_netif_get_ip_info(_mdns_get_esp_netif(packet->tcpip_if), &if_ip_info) == ESP_OK &&
                 memcmp(&if_ip_info.ip.addr, &packet->src.u_addr.ip4.addr, sizeof(esp_ip4_addr_t)) == 0) {
             return;
         }
-#if CONFIG_LWIP_IPV6
-    } else {
-        struct esp_ip6_addr if_ip6;
-        if (esp_netif_get_ip6_linklocal(_mdns_get_esp_netif(packet->tcpip_if), &if_ip6) == ESP_OK &&
-                memcmp(&if_ip6, &packet->src.u_addr.ip6, sizeof(esp_ip6_addr_t)) == 0) {
-            return;
-        }
-#endif
     }
+#endif /* CONFIG_MDNS_IPV4 */
+#if defined(CONFIG_MDNS_IPV4) && defined(CONFIG_MDNS_IPV6)
+    else
 #endif
```

Comment:
Nitpick: I would remove this part for readability, since the compiler would address this if we aim for optimizations

Reply:
Removed